src/webapp/components/customize-settings-dialog/CustomizeSettingsDialog.tsx

```diff
@@ -0,0 +1,215 @@
+            },
+            {
+                key: "import",
+                icon: <Icon>translate</Icon>,
+                text: i18n.t("Import JSON translations"),
+                onClick: () => {
+                    translationImportRef.current?.startImport();
+                    handleCloseMenu()
+                },
+            },
+        ],
+        [exportTranslation, loading, usecases]
+    );
+
+    const handleTranslationUpload = useCallback(
+        async (_key: string | undefined, lang: string, terms: Record<string, string>) => {
+            await importTranslation(() => usecases.config.importTranslations(lang, terms));
+        },
+        [usecases, importTranslation]
+    );
+
+    return (
+        <ConfirmationDialog
+            isOpen={true}
+            fullWidth={true}
+            onSave={save}
+            cancelText={i18n.t("Close")}
+            onCancel={onClose}
+            disableSave={disableSave}
+        >
+            <ImportTranslationDialog type="custom-text" ref={translationImportRef} onSave={handleTranslationUpload} />
+            <h3>{i18n.t("Home page logo")}</h3>
+            <Row>
+                <IconUpload>
+                    <IconContainer>
+                        <img src={logoVal} alt={`Home page logo`} />
+                    </IconContainer>
+                    <FileInput type="file" onChange={handleFileUpload} />
+                </IconUpload>
+            </Row>
+            <HeaderMenu>
+                <h3>{i18n.t("Customize application text")}</h3>{" "}
+                {!isCustomTextDefault && (
+                    <>
+                        <IconButton onClick={handleClickMenu}>
+                            <MoreVertIcon />
+                        </IconButton>
+                        <Menu anchorEl={menuAnchor} open={menuOpen} onClose={handleCloseMenu}>
+                            {menuAction.map(action => (
+                                <StyledMenuItem key={action.key} onClick={action.onClick}>
+                                    {action.icon}
+                                    {action.text}
+                                </StyledMenuItem>
+                            ))}
+                        </Menu>
+                    </>
+                )}
+            </HeaderMenu>
+
+            {customTextKeys.map(key => (
+                <Row key={key}>
+                    <TextField
+                        fullWidth={true}
+                        label={customTextLabel[key]}
+                        placeholder={i18n.t(defaultCustomText[key].referenceValue)}
+                        value={customTextVal[key] ? customTextVal[key]?.referenceValue : ""}
+                        InputLabelProps={{
+                            shrink: true,
+                        }}
+                        onChange={onChangeField(key)}
+                    />
+                </Row>
+            ))}
+        </ConfirmationDialog>
+    );
+};
+
+const Row = styled.div`
```

Comment:
UI comment: I would use `<Box marginBottom={3}/>` instead, because we end up having a lot of Wrappers, Row, Container, and things like that.